### ISSUE TITLE
attributes vs. properties

### ISSUE THREAD

**Issue by Scarygami** [PolymerElements/iron-component-page#32]
A recurring problem that people seem to have is the [new attribute to property mapping in 1.0](https://www.polymer-project.org/1.0/docs/devguide/properties.html#property-name-mapping).

They try something like `<custom-element myProperty="{{something}}">` because `myProperty` is listed in the element docs, where they should be using `<custom-element my-property="{{something}}">`

It would be nice to somehow display both the property and the attribute name in the generated docs. Especially beginners won't find their way all the way to the "property name mapping" part in the docs.

**Comment by atotic** [comment]
Gets me all the time, still. Great idea for linter mode.

**Comment by Scarygami** [comment]
Is this "Polymer linter" already available somewhere for testing/contributing or is it just an idea so far? :)

**Comment by atotic** [comment]
I think it is a work in progress

**Comment by arthurevans** [comment]
Adding this to the docs is also a good idea.

This and the auto-generated change events (property-changed) are subtleties that are obvious to the folks writing the API docs, but obscure to newcomers, especially if they go straight to the catalog.

**Comment by mbleigh** [comment]
My vote would be to have a toggle that flips "Properties" to "Attributes" and back, changing the formatting of the names but leaving everything else intact.

For `<prop>-changed` events, there are two ways I see to go about it -- have a toggle to show/hide them or have a separate section that just collects them into a comma-separated list. Any preference?

**Comment by atotic** [comment]
i'd love to see element generators, either as part of the demo, or a separate page. Nothing makes me happier as a developer than having an interactive playground.

**Comment by mbleigh** [comment]
Would be cool for sure, but I think out of scope for this issue ;)

**Comment by arthurevans** [comment]
For the events, I was thinking "Show/Hide generated."

Not sure about the flipping attributes/properties ... It makes it more compact but less discoverable (and possibly a little bit opaque).

**Comment by mbleigh** [comment]
Some elements have 10+ properties, meaning you're going to have a huge repeating section if you separate them out. A toggle helps reinforce that they're the same.

On further thinking, maybe it would be even better to have an "attribute name" along with the property name so there's no toggle needed. Show both at the same time for each one.

**Comment by arthurevans** [comment]
Yeah, I agree on not repeating them. Just adding an attribute name sounds good.

**Comment by arthurevans** [comment]
Because this behavior (and the property-changed events) are specific to Polymer, we should really add them to the autogenerated docs, so users on WC.org don't need to know Polymer-specific conventions to use the elements appropriately.

@aomarks WDYT? This is probably in the wrong repo, but would love to see this move forward.

**Comment by justinfagnani** [comment]
@arthurevans I need to check the analyzer output again, but we should be generating both an attribute record and property record, and we may be able to associate them because they have the same source range.

While the specific camel-case property name to kebab-case attribute name is from Polymer, I think the general idea that some attributes are associated with properties is general enough to keep in iron-component page, so we can link the docs of them.

**Comment by aomarks** [comment]
I think showing the attributes is a good idea, and https://github.com/PolymerElements/iron-doc-viewer/issues/125 is the same request.
